app/components/jwplayer-embed.js

```diff
@@ -0,0 +1,44 @@
+import Ember from 'ember';
+import jwPlayerAssets from '../modules/jwplayer-assets';
+
+const {RSVP, $, computed, Component} = Ember;
+
+export default Component.extend({
+	jwVideoDataUrl: 'https://cdn.jwplayer.com/v2/media/',
+
+	jwPlayerRootId: computed('media-id', function () {
+		return `jwPlayerTag${this.get('media-id')}`;
+	}),
+
+	/**
+	 * @returns {void}
+	 */
+	didInsertElement() {
+		RSVP.Promise.all([
+			jwPlayerAssets.load(),
+			this.getVideoData()
+		]).then(([, videoData]) => {
+			window.wikiaJWPlayer(
+				this.get('jwPlayerRootId'),
```

Comment:
If a user navigates away from the page before this is called, an error will be thrown. Check for `this.isDestroyed()`